tests/test_backups.py

```diff
@@ -16,105 +16,105 @@
 log = logging.getLogger(__file__)
 log.setLevel("DEBUG")
 
-CAPELLA_VERSION = os.getenv("CAPELLA_VERSION", "")
+pytestmark = pytest.mark.t4c_server
 
 
-def is_capella_5_x_x() -> bool:
-    return bool(re.match(r"5.[0-9]+.[0-9]+", CAPELLA_VERSION))
-
+@pytest.fixture(name="t4c_server_container_parametrized")
+def fixture_t4c_server_container_parametrized(
+    tmp_path: pathlib.Path, request
+) -> containers.Container:
+    env: dict[str, str] = {"REST_API_PASSWORD": "password"} | request.param
 
-def is_capella_6_x_x() -> bool:
-    return bool(re.match(r"6.[0-9]+.[0-9]+", CAPELLA_VERSION))
+    ports: dict[str, int | None] = {
+        "2036/tcp": None,
+        "8080/tcp": None,
+        "12036/tcp": None,
+    }
 
+    server_test_data_path = tmp_path / "test-server-data"
 
-@pytest.fixture(name="local_git_container")
-def fixture_local_git_container() -> containers.Container:
-    ports: dict[str, int | None] = {"80/tcp": None}
+    shutil.copytree(
+        src=pathlib.Path(__file__).parents[0]
+        / "t4c-server-test-data"
+        / "data"
+        / os.getenv("CAPELLA_VERSION", ""),
+        dst=server_test_data_path,
+    )
 
     with conftest.get_container(
-        image="local-git-server", ports=ports
+        image="t4c/server/server",
+        ports=ports,
+        environment=env,
+        path=server_test_data_path,
+        mount_path="/data",
     ) as container:
-        conftest.wait_for_container(container, "server started")
+        conftest.wait_for_container(
+            container, "!MESSAGE Warmup done for repository test-repo."
+        )
         yield container
 
 
-@pytest.fixture(name="t4c_backup_container")
-def fixture_t4c_backup_container(
-    local_git_container: containers.Container,
-    t4c_server_container: containers.Container,
-    request,
-) -> containers.Container:
-    t4c_server_container.reload()
-
-    t4c_repo_port = t4c_server_container.ports["2036/tcp"][0]["HostPort"]
-    t4c_cdo_port = t4c_server_container.ports["12036/tcp"][0]["HostPort"]
-
-    if "HTTP_PORT" in request.param:
-        request.param["HTTP_PORT"] = t4c_server_container.ports["8080/tcp"][0][
-            "HostPort"
-        ]
+@pytest.fixture(name="t4c_server_ports_parametrized")
+def fixture_t4c_server_ports_parametrized(
+    t4c_server_container_parametrized: containers.Container,
+) -> tuple[str, str, str]:
+    t4c_server_container_parametrized.reload()
+    return (
+        t4c_server_container_parametrized.ports["2036/tcp"][0]["HostPort"],
+        t4c_server_container_parametrized.ports["8080/tcp"][0]["HostPort"],
+        t4c_server_container_parametrized.ports["12036/tcp"][0]["HostPort"],
+    )
```

Comment:
Since I'm not able to run the tests at the moment, I can't check, but I'd probably leave it that way anyway. I'm pretty sure I tried to merge them but was unsuccessful, and it might take some time and if possible a slightly larger refactoring to merge them.

Reply:
Short example: 

```py
import pytest


@pytest.fixture(name="test")
def fixture_test(request: pytest.FixtureRequest):
    if not hasattr(request, "param"):
        return "no fixture param defined"

    return "fixture param defined: " + request.param


def test_test(test):
    print(test)


@pytest.mark.parametrize("test", [pytest.param("testvalue")], indirect=True)
def test_test_parameterized(test):
    print(test)
```

Output is: 

```
tests/test_test.py::test_test no fixture param defined
PASSED
tests/test_test.py::test_test_parameterized[testvalue] fixture param defined: testvalue
PASSED
```

This could remove a lot of code duplication.

Reply:
I have applied this now and as you mentioned, this eliminates a lot of duplication, but as mentioned on my part resulted in some major refactoring to make it the right way. I think the tests should be much clearer now and the `parametrized` fixtures are no longer needed.